packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -1059,93 +1191,89 @@ class _SelectToggleButtonRenderObject extends RenderShiftedBox {
+        case TextDirection.ltr:
+          childParentData.offset = Offset(leadingBorderSide.width, borderSide.width);
+          break;
+        case TextDirection.rtl:
+          childParentData.offset = Offset(trailingBorderSide.width, borderSide.width);
+          break;
+      }
+    } else {
+      switch (verticalDirection) {
+        case VerticalDirection.down:
+          childParentData.offset = Offset(borderSide.width, leadingBorderSide.width);
+          break;
+        case VerticalDirection.up:
+          childParentData.offset = Offset(borderSide.width, trailingBorderSide.width);
+          break;
+      }
     }
   }
 
-  double get _trailingBorderOffset => isLastButton ? trailingBorderSide.width : 0.0;
-
   Size _computeSize({required BoxConstraints constraints, required ChildLayouter layoutChild}) {
     if (child == null) {
-      return constraints.constrain(Size(
-        leadingBorderSide.width + trailingBorderSide.width,
-        horizontalBorderSide.width * 2.0,
-      ));
-    }
-
-    final double leftConstraint;
-    final double rightConstraint;
-
-    switch (textDirection) {
-      case TextDirection.ltr:
-        rightConstraint = _trailingBorderOffset;
-        leftConstraint = leadingBorderSide.width;
-
-        final BoxConstraints innerConstraints = constraints.deflate(
-          EdgeInsets.only(
-            left: leftConstraint,
-            top: horizontalBorderSide.width,
-            right: rightConstraint,
-            bottom: horizontalBorderSide.width,
-          ),
-        );
-
-        final Size childSize = layoutChild(child!, innerConstraints);
-
+      if (direction == Axis.horizontal) {
         return constraints.constrain(Size(
-          leftConstraint + childSize.width + rightConstraint,
-          horizontalBorderSide.width * 2.0 + childSize.height,
+          leadingBorderSide.width + trailingBorderSide.width,
+          borderSide.width * 2.0,
         ));
-      case TextDirection.rtl:
-        rightConstraint = leadingBorderSide.width;
-        leftConstraint = _trailingBorderOffset;
-
-        final BoxConstraints innerConstraints = constraints.deflate(
-          EdgeInsets.only(
-            left: leftConstraint,
-            top: horizontalBorderSide.width,
-            right: rightConstraint,
-            bottom: horizontalBorderSide.width,
-          ),
-        );
-
-        final Size childSize = layoutChild(child!, innerConstraints);
-
+      } else {
         return constraints.constrain(Size(
-          leftConstraint + childSize.width + rightConstraint,
-          horizontalBorderSide.width * 2.0 + childSize.height,
+          borderSide.width * 2.0,
+          leadingBorderSide.width + trailingBorderSide.width,
         ));
+      }
+    }
+
+    final double leftConstraint;
+    final double rightConstraint;
+    final double topConstraint;
+    final double bottomConstraint;
+
+    if (direction == Axis.horizontal) {
+      rightConstraint = trailingBorderSide.width;
```

Comment:
It does not matter that wether `left` or `right` is `trailingBorderSide.width`.

Reply:
Right -- since deflating the size constraints horizontally/vertically and the returned size accounts for the width of both sides. Maybe we should add a comment here to make it a little more obvious.